src/baconian.rs

```diff
@@ -0,0 +1,467 @@
+//! Bacon's cipher or the Baconian cipher is a method of steganography
+//! (a method of hiding a secret message as opposed to just a cipher) devised by Francis Bacon in 1605.
+//! A message is concealed in the presentation of text, rather than its content.
+//!
+//! Each character of the message plaintext is encoded as a 5-bit binary,
+//!  these are then "hidden" in a decoy message through the use of font variation.
+//!
+//! This cipher is very easy to crack, once the method of hiding is known, therefore this
+//! implementation includes the options to set whether the substitution is distinct for the whole
+//! alphabet, or whether it follows the classical method of treating 'I' and 'J', and 'U' and 'V'
+//! as interchangeable characters, as would have been the case in Bacon's time, though 'I' and 'V'
+//! were more common in text.
+//!
+//! Also, it allows the user to change the underlying binary
+//! character choice, this is traditionally 'a' and 'b', but optionally the user can choose any
+//! pair of characters.
+//!
+//! If no concealing text is given and boilerplate of "Loren ipsum..." is used, given the capacity
+//! to hide up to a 50 character plaintext.
+//!
+use std::collections::HashMap;
+use std::string::String;
+use common::cipher::Cipher;
+
+/// Default decoy plaintext
+const DEFAULT_DECOY: &str =
+    "Lorem ipsum dolor sit amet, ne tamquam eruditi splendide vix. \
+     Mea vitae latine philosophia in, et qui gubergren definiebas. \
+     Est et debet aliquam. Ei velit augue quo, quod veniam definitionem nam ut.";
+/// The default code length
+const CODE_LEN: usize = 5;
+
+/// Code mappings:
+///  * note: that str is preferred over char as it cannot be guaranteed that
+///     there will be a single codepoint for a given character.
+
+/// A traditional code set that makes 'I' = 'J' and 'V' = 'U'
+///     - as they had equivalent value in Bacon's day
+///     - generally, 'I' and 'V' were more common and used here
+lazy_static! {
+    static ref TRAD_CODES: HashMap<&'static str, &'static str> = hashmap!{
```

Comment:
For `TRAD_CODES` and `DISTINCT_CODES` hashmap, there seems to be a lot of overlap in the data stored. Surely there is a way we only need to define this information once?

Reply:
Agreed, I'll change it to that the outcome for the `TRADITIONAL` mapping is met, but using a `DISTINCT` codeset.